graphite_beacon/alerts.py

```diff
@@ -38,6 +40,59 @@ def __getitem__(self, index):
             return type(self)(islice(self, index.start, index.stop, index.step))
 
 
+class CronCallback(object):
+
+    """Callback that runs on a cron schedule."""
+
+    def __init__(self, callback, cron):
+        """Initialize a CronCallback object with the specified cron schedule and callback."""
+        self.callback = callback
+        self.cron = cron
+        self.is_running = False
+        self.handle = None
+
+    def start(self):
+        """Start running."""
+        if not self.is_running:
+            self.is_running = True
+            self.schedule_next_run()
+
+    def stop(self):
+        """Stop running."""
+        if self.is_running:
+            handle = self.handle
+            self.is_running = False
+            if handle:
+                ioloop.IOLoop.instance().remove_timeout(handle)
+                self.handle = None
+
+    def is_running(self):
+        """Is running."""
+        return self.is_running
+
+    def scheduled_run(self):
+        """Invoke the callback and schedule the next run."""
+        if self.is_running:
+            LOGGER.debug("CronCallback: running cron schedule")
+            try:
+                self.callback()
+            finally:
+                self.schedule_next_run()
+
+    def schedule_next_run(self):
+        """Schedule the next run of this callback."""
+        if self.is_running:
+            now = datetime.now()
+            next_time = self.cron.get_next(datetime)
+            while next_time <= now:
```

Comment:
Do you think it's worth having a limit on how far this will search before erroring/failing early?